diff --git a/tools/Makefile b/tools/Makefile
index 36269196..d55d8698 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -267,7 +267,16 @@ HOSTCFLAGS_sha512.o := -pedantic -DCONFIG_SHA512 -DCONFIG_SHA384
 quiet_cmd_wrap = WRAP    $@
 cmd_wrap = echo "\#include <../$(patsubst $(obj)/%,%,$@)>" >$@
 
-$(obj)/boot/%.c $(obj)/common/%.c $(obj)/env/%.c $(obj)/lib/%.c:
+$(obj)/boot/%.c: tools/boot/%.c
+	$(call cmd,wrap)
+
+$(obj)/common/%.c: tools/common/%.c
+	$(call cmd,wrap)
+
+$(obj)/env/%.c: tools/env/%.c
+	$(call cmd,wrap)
+
+$(obj)/lib/%.c: tools/lib/%.c
 	$(call cmd,wrap)
 
 clean-dirs := lib common
